parcels/fieldset.py

```diff
@@ -343,3 +343,97 @@ def advancetime(self, fieldset_new):
                 advance = advance2
                 gnew.advanced = True
             f.advancetime(fnew, advance == 1)
+
+    def computeTimeChunk(self, time, dt):
+        signdt = np.sign(dt)
+        nextTime = np.infty * signdt
+
+        for g in self.gridset.grids:
+            g.advanced = 0
+        for f in self.fields:
+            if f.name == 'UV' or not f.grid.time_partial_load:
+                continue
+            g = f.grid
+            nextTime_loc = np.infty * signdt
+            if g.advanced == 0:
+                if g.ti == -1:
+                    g.time = g.time_full
+                    g.tdim = 3
+                    g.ti, _ = f.time_index(time)
+                    if g.ti > 0 and signdt == -1:
+                        if g.ti == len(g.time_full)-1:
+                            g.ti -= 2
+                        else:
+                            g.ti -= 1
+                    g.time = g.time_full[g.ti:g.ti+3]
+                    g.advanced = 2
+                else:
+                    if signdt >= 0 and time >= g.time[1] and g.ti < len(g.time_full)-3:
+                        g.ti += 1
+                        g.time = g.time_full[g.ti:g.ti+3]
+                        g.advanced = 1
+                        if g.ti != len(g.time_full)-3:
+                            nextTime_loc = g.time[2]
+                    if signdt == -1 and time <= g.time[1] and g.ti > 0:
+                        g.ti -= 1
+                        g.time = g.time_full[g.ti:g.ti+3]
+                        g.advanced = 1
```

Comment:
This code is quite long and confusing. Can be cleaned up and pythonized a bit? E.g. both if statements have `g.advanced=1`, so can be taken out of if loops?